### PR TITLE
added spring and hibernate frameworks to the project;  implemented Us…

### PR DESCRIPTION
**Pull request by MaliukDaria**
…erDao and UserService; configured ApplicationContext and DataSource

### CODE REVIEW

**Review by IrynaDim — APPROVED** [review]

File: src/main/java/com/spring/test/config/AppConfig.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import org.springframework.context.annotation.Configuration;
+import org.springframework.orm.hibernate5.LocalSessionFactoryBean;
+
+@Configuration
+@ComponentScan(basePackages = {
+        "com.spring.test.dao",
+        "com.spring.test.service"
+})
+public class AppConfig {
+    public DataSource getDataSource() {
```
Comment (Sofasmile) [inline src/main/java/com/spring/test/config/AppConfig.java:19]: let's add Bean annotation

File: src/main/java/com/spring/test/dao/UserDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class UserDaoImpl implements UserDao {
+    @Autowired
+    private SessionFactory sessionFactory;
```
Comment (Sofasmile) [inline src/main/java/com/spring/test/dao/UserDaoImpl.java]: What is better: setter auto wiring, constructor auto wiring, or field injection?

File: src/main/java/com/spring/test/service/UserServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import com.spring.test.dao.UserDao;
+import com.spring.test.model.User;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class UserServiceImpl implements UserService {
+    @Autowired
+    private UserDao userDao;
```
Comment (Sofasmile) [inline src/main/java/com/spring/test/service/UserServiceImpl.java]: What is better: setter auto wiring, constructor auto wiring, or field injection?

File: src/main/java/com/spring/test/config/AppConfig.java (hunk 1 above)
Comment (MaliukDaria) [reply src/main/java/com/spring/test/config/AppConfig.java:19]: done

File: src/main/java/com/spring/test/dao/UserDaoImpl.java (hunk 2 above)
Comment (MaliukDaria) [reply src/main/java/com/spring/test/dao/UserDaoImpl.java]: constructor

File: src/main/java/com/spring/test/service/UserServiceImpl.java (hunk 3 above)
Comment (MaliukDaria) [reply src/main/java/com/spring/test/service/UserServiceImpl.java]: constructor

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**